remote-api/src/test/java/org/alfresco/rest/api/impl/TagsImplTest.java

```diff
@@ -302,4 +314,34 @@ private static Tag createTagWithNodeRef(final String tagName)
             .tag(tagName)
             .create();
     }
+
+    @Test
+    public void testAddTagsToNode()
+    {
+        final List<String> tagNames = List.of("tag1","tag2");
+        final List<Tag> tagsToCreate = createTags(tagNames);
+        given(taggingServiceMock.addTags(any(), any())).willAnswer(invocation -> createTagAndNodeRefPairs(invocation.getArgument(1)));
+        final List<Tag> actualCreatedTags = objectUnderTest.addTags(nodesMock.getNode(any()).getNodeId(),tagsToCreate);
+        then(taggingServiceMock).should().addTags(TAG_NODE_REF, tagNames);
+        final List<Tag> expectedTags = createTagsWithNodeRefs(tagNames);
+        assertThat(actualCreatedTags)
+                .isNotNull()
+                .isEqualTo(expectedTags);
+    }
+
+    @Test
+    public void testAddTagsToNodeWithResponseNotIndexed()
+    {
+        final List<String> tagNames = List.of("tag1","tag2");
+        final List<Tag> tagsToCreate = createTags(tagNames);
+        given(taggingServiceMock.addTags(any(), any())).willAnswer(invocation -> createTagAndNodeRefPairs(invocation.getArgument(1)));
+        given(taggingServiceMock.findTaggedNodesAndCountByTagName(any())).willReturn(Collections.emptyList());
+        final List<Tag> actualCreatedTags = objectUnderTest.addTags(nodesMock.getNode(any()).getNodeId(),tagsToCreate);
+        then(taggingServiceMock).should().addTags(TAG_NODE_REF, tagNames);
+        final List<Tag> expectedTags = createTagsWithNodeRefs(tagNames);
+        assertThat(actualCreatedTags)
+                .isNotNull()
+                .isEqualTo(expectedTags);
```

Comment:
This test should not be passing, because the count was not set in the tags. Please look to see if we can reasonably include the count in the `equals()` method of `Tag`.

Reply:
Nice catch Tom! 👍